ns_api.py

```diff
@@ -407,11 +407,12 @@ def __init__(self, part_dict=None):
         if 'actualDepartureTrack' in part_dict:
             self.actual_platform = part_dict['actualDepartureTrack']
             self.platform_changed = bool(self.actual_platform != self.planned_platform)
-        if self.actual_time is not None:
-            try:
-                self.delay = self.actual_time - self.planned_time
-            except KeyError:
-                self.delay = None
+
+        if self.actual_time is not None and self.planned_time is not None:
```

Comment:
Do you really have to state 'is not None', can't you just say 

`if self.actual_time and self.planned_time:`  ?

Reply:
As the old code already did it this way, I'll merge for now; we can always refactor and test later :)